### DISCUSSION TITLE
Errors in MET

### DISCUSSION THREAD

**Discussion by Kuriyama2000** [dtcenter/METplus#2646] [Configuration] [answered]
My name is William and I am a master's student at the National Institute for Space Research in Brazil.

I'm using MET to develop it, however, I'm getting some errors when I try to run "Grid-Stat".

I'm trying to run the Grid-Stat function, for this I have 2 files (Prev and Obs - NetCDF) and within them I want to analyze the variable t2m (temperature) which is structured as follows:

william@probook:~/DTC/MET-11.1.1/bin$ ncdump -h [t2m_test_obs.nc](http://t2m_test_obs.nc/)
netcdf t2m_test_obs {
dimensions:
longitude = 38 ;
latitude = 19 ;
time = 2208 ;
variables:
float longitude(longitude) ;
longitude:units = "degrees_east" ;
longitude:long_name = "longitude" ;
float latitude(latitude) ;
latitude:units = "degrees_north" ;
latitude:long_name = "latitude" ;
int time(time) ;
time:units = "hours since 1900-01-01 00:00:00.0" ;
time:long_name = "time" ;
time:calendar = "gregorian" ;
short t2m(time, latitude, longitude) ;
t2m:scale_factor = 0.000624629391155983 ;
t2m:add_offset = 289.737602724367 ;
t2m:_FillValue = -32767s ;
t2m:missing_value = -32767s ;
t2m:units = "K" ;
t2m:long_name = "2 metre temperature" ;

With the information from NetCDF I edited the MET configuration file as follows (GridStatConfig):

// Forecast and observation fields to be verified
//

fcst = {
   field = [
  {
      name   = "t2m";
      level   = [ "* , * , *" ];
      cat_thresh = [ >0.0, >=5.0 ];
  }
   ];
}

obs = {
   field = [
  {
      name   = "t2m";
      level   = ["* , * , *" ];
      cat_thresh = [ >0.0, >=5.0 ];
  }
   ];
}

// Output observation type to be written
obtype = "ANALYS";

// Verification grid
regrid = {
   to_grid = NONE;
   method = NEAREST;
   width   = 1;
   vld_thresh = 0.5;
   shape   = SQUARE;
}

However, when I give the command to run grid-stat, it gives the following error:

./grid_stat [t2m_test_prev.nc](http://t2m_test_prev.nc/) [t2m_test_obs.nc](http://t2m_test_obs.nc/) GridStatConfig

ERROR  : NcCfFile::getData(NcVar *, const LongArray &, DataPlane &) const -> needed 3 arguments for variable t2m, got 2

In this case, it asks for 3 variables for the t2m, and is only recognizing 2. I've already tried changing the "level" by "Time, latitude, distance" and also by height levels, however, the error remains, I looked for something related in the guide user however, I didn't find anything about it.

Thank you in advance for your attention!

**Comment by JohnHalleyGotway** [comment, marked as answer]
Hi, I see you have a question about configuring Grid-Stat. Thanks for sending along the details of your NetCDF file and config file entries. Grid-Stat is designed to be run once per output time. Your data has 3 dimensions: time, latitude, and longitude:
```
short t2m(time, latitude, longitude) ;
```
As the error message states, you should have exactly two * characters to in the level string in the slots for the lat/lon dimensions.
```
name = "t2m";
level = ["(0,*,*)" ];
```
In this example, I just put a 0 for the first dimension. That means just use data for the first time slice. Although I see that you have 2208 time slices in this file (`time = 2208 ;`). Instead of specifying an integer index into that time dimension, you can instead specify a time string similar to this `level = ["(20240807_00,*,*)" ];`. The timestrings can specified in YYYYMMDD_HHMMSS format or YYYYMMDD or YYYYMMDD_HH.

Two additional suggestions:
1. Whenever getting going with new datasets, I always recommend starting with the plot_data_plane tool. That just reads a single slice of gridded data from your file and plots it. That'll help figure out what config strings to use and confirm that MET's placing your data on the correct place on the earth.
```
plot_data_plane t2m_test_obs.nc t2m.ps 'name="t2m"; level="(0,*,*)";'
```
2. While it's totally fine to configure and run the MET tools directly yourself, you may find it tedious to write a script to loop over all 2208 times in that file. That's a good motivation for running the METplus wrappers to automate calls to the MET tools. You can take a look at the existing [METplus Use Cases](https://metplus.readthedocs.io/en/latest/generated/model_applications/index.html). With METplus you can define config options for multiple MET tools in a single configuration file... and you can easily loop over multiple times and easily define the input files for each time.

Please let me know how it goes. If you get additional errors, you could consider posting sample data and config files to our anonymous ftp site following [these instructions](https://github.com/dtcenter/METplus/discussions/954).
